crates/holochain_websocket/tests/integration.rs

```diff
@@ -452,3 +481,54 @@ async fn cancel_response() {
     c_jh.await.unwrap();
     s_jh.await.unwrap();
 }
+
+#[tokio::test(flavor = "multi_thread")]
+#[cfg(feature = "slow_tests")]
+#[ignore = "takes at least 45s, please run me occasionally"]
+async fn can_handle_many_connections_and_disconnects() {
+    holochain_trace::test_run().ok();
+    let (handle, listener) = server().await;
+    let s_jh = server_recv_multi(listener);
+    let binding = handle.local_addr().clone();
+
+    let b2 = binding.clone();
+    tokio::spawn(async move {
+        let mut senders = Vec::new();
+        for i in 0..100_000 {
+            if i % 800 == 0 {
+                senders.clear();
+            }
+
+            let (mut sender, _) = connect(b2.clone(), Arc::new(WebsocketConfig::default()))
+                .instrument(tracing::debug_span!("client"))
+                .await
+                .unwrap();
+
+            sender
+                .signal(TestString("Hey from client".to_owned()))
+                .instrument(tracing::debug_span!("client_sending_message"))
+                .await
+                .unwrap();
+
+            senders.push(sender);
+        }
+
+        senders.clear();
```

Comment:
I don't understand why you clear the senders vector every 800 times. Or rather, why you have that vector to begin with.

Reply:
That's a fair point, it's not very clear what the intention of that is and it deserves a code comment. There is a limit on how many connections we can open and a test could be added to test for that but I don't want to test for resource exhaustion here, what I care about is whether the code correctly releases resources when senders are dropped and can keep accepting new connections. 800 is rather arbitrary, it's double the limit that Holochain's admin websocket enforces

Reply:
Done